src/main/java/forestry/factory/recipes/CarpenterRecipe.java

```diff
@@ -55,4 +64,28 @@ public FluidStack getFluidResource() {
 	public IDescriptiveRecipe getCraftingGridRecipe() {
 		return internal;
 	}
+
+	@Override
+	public ResourceLocation getId() {
+		return id;
+	}
+
+	// TODO: Remove ShapedRecipeCustom. We cannot serialize this without ShapedRecipeCustom having a serializer
```

Comment:
Is ShapedRecipeCustom still needed at all anymore ?

Reply:
I'm pretty sure it's not, so I'll start removing it